CRM/Core/BAO/UFGroup.php

```diff
@@ -1451,10 +1458,19 @@ public static function add(&$params, $ids = []) {
     if (!empty($params['group_type']) && is_array($params['group_type'])) {
       $params['group_type'] = implode(',', $params['group_type']);
     }
+
+    $ufGroupID = CRM_Utils_Array::value('ufgroup', $ids, CRM_Utils_Array::value('id', $params));
```

Comment:
If possible can you squash this down - like we do on GroupContact

```
 $hook = empty($params['id']) ? 'create' : 'edit';
    CRM_Utils_Hook::pre($hook, 'GroupContact', ($params['id'] ?? NULL) , $params);
```

Note it's OK not to check the $ids because that param can be deprecated - I'll put up a PR for that

Reply:
https://github.com/civicrm/civicrm-core/pull/19060